### DISCUSSION TITLE
HTTP to Flask

### DISCUSSION THREAD

**Discussion by hana-gharrad** [gama-platform/gama#50] [Q&A] [answered]
Hello,

I'm trying the network agent provided in "HTTP Post.gaml" file (Network). However for some reason I receive only the code. I tried to display all what it is fetched using:

reflex get_message {
		loop while:has_more_message()
		{
			//read a message
			message mess <- fetch_message();
                        write mess;
             }
}
but I see only the code which is code = 200, I tried postman to see the result of the request and I'm able to receive the body.
I also used wireshark to see if gama is sending the request and I can see the package returned.
I did not understand why I can receive only the code without headers or body.
Any help will be appreciated.

**Note**: The server is located in local machine (under Windows Subsystem for Linux) and I tried gama 1.9.0 and 1.9.1 and in both I have the same issue.

Many thanks,
Kind regards

**Comment by benoitgaudou** [comment, marked as answer]
Dear,
I just tried the example models in GAMA 1.9.1 and in particular HTTP
POST.gaml (see below), with the Flask server provided in the scripts folder.
And I get not only the code but also other information.

Can you try it to check you get what you are expected ?

Cheers

Benoit


[image: Capture d’écran 2023-09-06 à 07.39.55.png]

Le mar. 5 sept. 2023 à 15:38, hana-gharrad ***@***.***> a
écrit :

**Reply by hana-gharrad** [reply]
Thank you for the reply @benoitgaudou . I seems that I need to use jsonify to format the reply, it return only well formatted reply otherwise it will return nil.

Thanks again.
